cli/cmd/alert_rules.go

```diff
@@ -0,0 +1,372 @@
+		Long:  "Creates a new single alert rule.",
+		RunE: func(_ *cobra.Command, args []string) error {
+			if !cli.InteractiveMode() {
+				return errors.New("interactive mode is disabled")
+			}
+
+			response, err := promptCreateAlertRule()
+			if err != nil {
+				return errors.Wrap(err, "unable to create alert rule")
+			}
+
+			cli.OutputHuman(fmt.Sprintf("The alert rule was created with GUID %s \n", response.Data.Guid))
+			return nil
+		},
+	}
+)
+
+func init() {
+	// add the alert-rule command
+	rootCmd.AddCommand(alertRulesCommand)
+
+	// add sub-commands to the alert-rule command
+	alertRulesCommand.AddCommand(alertRulesListCommand)
+	alertRulesCommand.AddCommand(alertRulesShowCommand)
+	alertRulesCommand.AddCommand(alertRulesCreateCommand)
+	alertRulesCommand.AddCommand(alertRulesDeleteCommand)
+}
+
+func buildAlertRuleDetailsTable(rule api.AlertRule) string {
+	var (
+		details     [][]string
+		updatedTime string
+	)
+	severities := api.NewAlertRuleSeveritiesFromIntSlice(rule.Filter.Severity).ToStringSlice()
+
+	if nano, err := strconv.ParseInt(rule.Filter.CreatedOrUpdatedTime, 10, 64); err == nil {
+		updatedTime = time.Unix(nano/1000, 0).Format(time.RFC3339)
+	}
+	details = append(details, []string{"SEVERITIES", strings.Join(severities, ", ")})
+	details = append(details, []string{"EVENT CATEGORIES", strings.Join(rule.Filter.EventCategories, ", ")})
+	details = append(details, []string{"DESCRIPTION", rule.Filter.Description})
+	details = append(details, []string{"UPDATED BY", rule.Filter.CreatedOrUpdatedBy})
+	details = append(details, []string{"LAST UPDATED", updatedTime})
+
+	detailsTable := &strings.Builder{}
+	detailsTable.WriteString(renderOneLineCustomTable("ALERT RULE DETAILS",
+		renderCustomTable([]string{}, details,
+			tableFunc(func(t *tablewriter.Table) {
+				t.SetBorder(false)
+				t.SetColumnSeparator(" ")
+				t.SetAutoWrapText(false)
+				t.SetAlignment(tablewriter.ALIGN_LEFT)
+			}),
+		),
+		tableFunc(func(t *tablewriter.Table) {
+			t.SetBorder(false)
+			t.SetAutoWrapText(false)
+		}),
+	),
+	)
+
+	if len(rule.Channels) > 0 {
+		channels := [][]string{{strings.Join(rule.Channels, "\n")}}
+		detailsTable.WriteString(renderCustomTable([]string{"CHANNELS"}, channels,
+			tableFunc(func(t *tablewriter.Table) {
+				t.SetBorder(false)
+				t.SetColumnSeparator(" ")
+			}),
+		),
+		)
+		detailsTable.WriteString("\n")
+	}
+
+	if len(rule.Filter.ResourceGroups) > 0 {
+		resourceGroups := [][]string{{strings.Join(rule.Filter.ResourceGroups, "\n")}}
+		detailsTable.WriteString(renderCustomTable([]string{"RESOURCE GROUPS"}, resourceGroups,
+			tableFunc(func(t *tablewriter.Table) {
+				t.SetBorder(false)
+				t.SetColumnSeparator(" ")
+			}),
+		),
+		)
+	}
+
+	return detailsTable.String()
+}
+
+func promptCreateAlertRule() (api.AlertRuleResponse, error) {
+	channelList, channelMap := getAlertChannels()
+
+	if len(channelList) < 1 {
+		return api.AlertRuleResponse{}, errors.New("No Alert Channels found.")
```

Comment:
I am not sure I like the capitalization in this error message:
```suggestion
		return api.AlertRuleResponse{}, errors.New("no alert channels found")
```